docs/conf.py

```diff
@@ -65,10 +68,56 @@
 
 autodoc_typehints_format = "short"
 
-exclude_patterns = ["_build"]
+autodoc_typehints = "description"
 
-autodoc_typehints = "signature"
-
-autosummary_generate = False
+autosummary_generate = True
 
 epub_show_urls = "footnote"
+
+autodoc_type_aliases = {
+    "jaxsim.typing.PyTree": "jaxsim.typing.PyTree",
+    "jaxsim.typing.Vector": "jaxsim.typing.Vector",
+    "jaxsim.typing.Matrix": "jaxsim.typing.Matrix",
+    "jaxsim.typing.Array": "jaxsim.typing.Array",
+    "jaxsim.typing.Int": "jaxsim.typing.Int",
+    "jaxsim.typing.Bool": "jaxsim.typing.Bool",
+    "jaxsim.typing.Float": "jaxsim.typing.Float",
+    "jaxsim.typing.ScalarLike": "jaxsim.typing.ScalarLike",
+    "jaxsim.typing.ArrayLike": "jaxsim.typing.ArrayLike",
+    "jaxsim.typing.VectorLike": "jaxsim.typing.VectorLike",
+    "jaxsim.typing.MatrixLike": "jaxsim.typing.MatrixLike",
+    "jaxsim.typing.IntLike": "jaxsim.typing.IntLike",
+    "jaxsim.typing.BoolLike": "jaxsim.typing.BoolLike",
+    "jaxsim.typing.FloatLike": "jaxsim.typing.FloatLike",
+}
```

Comment:
Can you add a comment in the code about the reasons why this is necessary?

Reply:
Done in https://github.com/ami-iit/jaxsim/pull/187/commits/8906c9c7cb962fbc3324b167e866b09186f3fa8b